src/server/storage.zig

```diff
@@ -121,25 +127,19 @@ pub const RegionFile = struct { // MARK: RegionFile
 			return;
 		}
 
-		const data = main.stackAllocator.alloc(u8, totalSize + headerSize);
-		defer main.stackAllocator.free(data);
-		var i: usize = 0;
-		std.mem.writeInt(u32, data[i..][0..4], version, .big);
-		i += 4;
-		std.mem.writeInt(u32, data[i..][0..4], @intCast(totalSize), .big);
-		i += 4;
-		for(0..regionVolume) |j| {
-			std.mem.writeInt(u32, data[i..][0..4], @intCast(self.chunks[j].len), .big);
-			i += 4;
-		}
-		std.debug.assert(i == headerSize);
+		var writer = BinaryWriter.initCapacity(main.stackAllocator, .big, totalSize + headerSize);
+		defer writer.deinit();
 
-		for(0..regionVolume) |j| {
-			const size = self.chunks[j].len;
-			@memcpy(data[i..][0..size], self.chunks[j]);
-			i += size;
+		writer.writeInt(u32, version);
+		writer.writeInt(u32, @truncate(totalSize));
```

Comment:
This could potentially hide bugs, better keep intCast, so we at least get a crash from it.

Reply:
```suggestion
		writer.writeInt(u32, @intCast(totalSize));
```